src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -168,22 +152,21 @@ public HttpResponse getConsoleText(StaplerRequest2 req, StaplerResponse2 rsp) th
      */
     @GET
     @WebMethod(name = "consoleOutput")
-    public HttpResponse getConsoleOutput(StaplerRequest2 req) throws IOException {
+    public void getConsoleOutput(StaplerRequest2 req, StaplerResponse2 res) throws IOException {
         String nodeId = req.getParameter("nodeId");
         if (nodeId == null) {
             logger.error("'consoleJson' was not passed 'nodeId'.");
-            return HttpResponses.errorJSON("Error getting console json");
+            res.getWriter().write("Error getting console json");
         }
         logger.debug("getConsoleOutput was passed node id '{}'.", nodeId);
         // This will be a step, so return it's log output.
         // startByte to start getting data from. If negative will startByte from end of string with
         // LOG_THRESHOLD.
         Long startByte = parseIntWithDefault(req.getParameter("startByte"), -LOG_THRESHOLD);
-        JSONObject data = getConsoleOutputJson(nodeId, startByte);
-        if (data == null) {
-            return HttpResponses.errorJSON("Something went wrong - check Jenkins logs.");
+        AnnotatedLargeText<? extends FlowNode> logText = getLogForNode(nodeId);
+        if (logText != null) {
+            logText.writeLogTo(0, res.getOutputStream());
```

Comment:
likely need to pass offsets along as this will probably overload the browser with too much to go in the DOM?